reference/conanfile/tools/files/basic.rst

```diff
@@ -138,6 +138,31 @@ Parameters:
     - **encoding** (Optional, Defaulted to ``utf-8``): Specifies the input and output files text encoding.
 
 
+conan.tools.files.rm()
+----------------------
+
+.. code-block:: python
+
+    def rm(conanfile, pattern, folder, recursive=False)
+
+
+Remove the files following the ``pattern`` (fnmatch) from the specified ``folder``.
+
+.. code-block:: python
+
+    from conan.tools.files import rm
+
+    rm(self, "*.tmp", self.build_folder, recursive=True)
+
+
+Parameters:
+    - **conanfile**: Conanfile object.
+    - **pattern** (Required): Pattern that the files to be removed have to match (fnmatch).
+    - **folder**: Folder to search/remove the files.
+    - **recursive**: If ``recursive`` is specified it will search in the subfolders.
```

Comment:
Is it worth putting `Required` or `Optional` for every parameter? I mean I think it's weird to put only one of them.

```
    - **conanfile** (Required): Conanfile object.
    - **pattern** (Required): Pattern that the files to be removed have to match (fnmatch).
    - **folder** (Required): Folder to search/remove the files.
    - **recursive** (Optional): If ``recursive`` is specified it will search in the subfolders.
```

Reply:
Yeah, I would probably define as ``(optional)`` only the optional ones, and maybe not even that, as the signature is there and it is visible what arguments have defaults

Reply:
Removed optional and required in the whole module.